[Bug]: Blank page at login screen in 0.30.0 version

### Installation Method

Docker

### The Problem

Blank page at login screen

### Version of Stirling-PDF

0.30.0

### Last Working Version of Stirling-PDF

0.29.0

### Page Where the Problem Occurred

http://localhost:8181/login

### Docker Configuration

```txt
version: '3.3'
services:
  stirling-pdf:
    image: frooodle/s-pdf:latest
    restart: unless-stopped
    ports:
      - '127.0.0.1:8181:8080'
    volumes:
      - ./trainingData:/usr/share/tessdata #Required for extra OCR languages
      - ./extraConfigs:/configs
#      - /location/of/customFiles:/customFiles/
#      - /location/of/logs:/logs/
    environment:
      - DOCKER_ENABLE_SECURITY=true
      - SECURITY_ENABLE_LOGIN=true
      - SECURITY_INITIALLOGIN_USERNAME=username
      - SECURITY_INITIALLOGIN_PASSWORD=xxxxx
      - INSTALL_BOOK_AND_ADVANCED_HTML_OPS=false
      - LANGS=fr_FR
```


### Relevant Log Output

```txt
stirling-pdf-1  | 07:57:35.396 [qtp1397904249-33] WARN  o.e.j.ee10.servlet.ServletChannel - /login
stirling-pdf-1  | jakarta.servlet.ServletException: Request processing failed: org.thymeleaf.exceptions.TemplateProcessingException: Error during execution of processor 'org.thymeleaf.spring6.processor.SpringActionTagProcessor' (template: "login" - line 137, col 77)
stirling-pdf-1  |       at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1022)
stirling-pdf-1  |       at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:903)
stirling-pdf-1  |       at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:527)
stirling-pdf-1  |       at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885)
stirling-pdf-1  |       at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:614)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHolder.handle(ServletHolder.java:736)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1614)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.websocket.servlet.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:195)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at stirling.software.SPDF.config.security.IPRateLimitingFilter.doFilter(IPRateLimitingFilter.java:41)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at stirling.software.SPDF.config.MetricsFilter.doFilterInternal(MetricsFilter.java:48)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at stirling.software.SPDF.config.security.UserBasedRateLimitingFilter.doFilterInternal(UserBasedRateLimitingFilter.java:48)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:110)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:108)
stirling-pdf-1  |       at org.springframework.security.web.FilterChainProxy.lambda$doFilterInternal$3(FilterChainProxy.java:231)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$FilterObservation$SimpleFilterObservation.lambda$wrap$1(ObservationFilterChainDecorator.java:479)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$AroundFilterObservation$SimpleAroundFilterObservation.lambda$wrap$1(ObservationFilterChainDecorator.java:340)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator.lambda$wrapSecured$0(ObservationFilterChainDecorator.java:82)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:128)
stirling-pdf-1  |       at org.springframework.security.web.access.intercept.AuthorizationFilter.doFilter(AuthorizationFilter.java:100)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:126)
stirling-pdf-1  |       at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:120)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:131)
stirling-pdf-1  |       at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:85)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:100)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:145)
stirling-pdf-1  |       at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:101)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:179)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:151)
stirling-pdf-1  |       at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:129)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at stirling.software.SPDF.config.security.FirstLoginFilter.doFilterInternal(FirstLoginFilter.java:43)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:227)
stirling-pdf-1  |       at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:221)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at stirling.software.SPDF.config.security.IPRateLimitingFilter.doFilter(IPRateLimitingFilter.java:41)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:107)
stirling-pdf-1  |       at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:93)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.csrf.CsrfFilter.doFilterInternal(CsrfFilter.java:117)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90)
stirling-pdf-1  |       at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.context.SecurityContextHolderFilter.doFilter(SecurityContextHolderFilter.java:82)
stirling-pdf-1  |       at org.springframework.security.web.context.SecurityContextHolderFilter.doFilter(SecurityContextHolderFilter.java:69)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:62)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.session.DisableEncodeUrlFilter.doFilterInternal(DisableEncodeUrlFilter.java:42)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$AroundFilterObservation$SimpleAroundFilterObservation.lambda$wrap$0(ObservationFilterChainDecorator.java:323)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:224)
stirling-pdf-1  |       at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
stirling-pdf-1  |       at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:233)
stirling-pdf-1  |       at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:191)
stirling-pdf-1  |       at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:113)
stirling-pdf-1  |       at org.springframework.web.servlet.handler.HandlerMappingIntrospector.lambda$createCacheFilter$3(HandlerMappingIntrospector.java:195)
stirling-pdf-1  |       at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:113)
stirling-pdf-1  |       at org.springframework.web.filter.CompositeFilter.doFilter(CompositeFilter.java:74)
stirling-pdf-1  |       at org.springframework.security.config.annotation.web.configuration.WebMvcSecurityConfiguration$CompositeFilterChainProxy.doFilter(WebMvcSecurityConfiguration.java:230)
stirling-pdf-1  |       at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:352)
stirling-pdf-1  |       at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:268)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at org.springframework.web.filter.ServerHttpObservationFilter.doFilterInternal(ServerHttpObservationFilter.java:113)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
stirling-pdf-1  |       at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.FilterHolder.doFilter(FilterHolder.java:205)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1586)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler$MappedServlet.handle(ServletHandler.java:1547)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletChannel.dispatch(ServletChannel.java:824)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletChannel.handle(ServletChannel.java:436)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletHandler.handle(ServletHandler.java:464)
stirling-pdf-1  |       at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:575)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.SessionHandler.handle(SessionHandler.java:703)
stirling-pdf-1  |       at org.eclipse.jetty.server.handler.ContextHandler.handle(ContextHandler.java:1060)
stirling-pdf-1  |       at org.eclipse.jetty.server.Server.handle(Server.java:181)
stirling-pdf-1  |       at org.eclipse.jetty.server.internal.HttpChannelState$HandlerInvoker.run(HttpChannelState.java:661)
stirling-pdf-1  |       at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:406)
stirling-pdf-1  |       at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:322)
stirling-pdf-1  |       at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:99)
stirling-pdf-1  |       at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
stirling-pdf-1  |       at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:478)
stirling-pdf-1  |       at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:441)
stirling-pdf-1  |       at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:293)
stirling-pdf-1  |       at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:201)
stirling-pdf-1  |       at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:311)
stirling-pdf-1  |       at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:979)
stirling-pdf-1  |       at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1209)
stirling-pdf-1  |       at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1164)
stirling-pdf-1  |       at java.base/java.lang.Thread.run(Thread.java:1583)
stirling-pdf-1  | Caused by: org.thymeleaf.exceptions.TemplateProcessingException: Error during execution of processor 'org.thymeleaf.spring6.processor.SpringActionTagProcessor' (template: "login" - line 137, col 77)
stirling-pdf-1  |       at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:117)
stirling-pdf-1  |       at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95)
stirling-pdf-1  |       at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633)
stirling-pdf-1  |       at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1314)
stirling-pdf-1  |       at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205)
stirling-pdf-1  |       at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:136)
stirling-pdf-1  |       at org.thymeleaf.engine.TemplateManager.parseAndProcess(TemplateManager.java:592)
stirling-pdf-1  |       at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1103)
stirling-pdf-1  |       at org.thymeleaf.TemplateEngine.process(TemplateEngine.java:1077)
stirling-pdf-1  |       at org.thymeleaf.spring6.view.ThymeleafView.renderFragment(ThymeleafView.java:372)
stirling-pdf-1  |       at org.thymeleaf.spring6.view.ThymeleafView.render(ThymeleafView.java:192)
stirling-pdf-1  |       at org.springframework.web.servlet.DispatcherServlet.render(DispatcherServlet.java:1432)
stirling-pdf-1  |       at org.springframework.web.servlet.DispatcherServlet.processDispatchResult(DispatcherServlet.java:1168)
stirling-pdf-1  |       at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1106)
stirling-pdf-1  |       at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979)
stirling-pdf-1  |       at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014)
stirling-pdf-1  |       ... 167 common frames omitted
stirling-pdf-1  | Caused by: java.lang.IllegalStateException: Response is committed
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletContextRequest.getSession(ServletContextRequest.java:477)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletApiRequest.getSession(ServletApiRequest.java:519)
stirling-pdf-1  |       at org.eclipse.jetty.ee10.servlet.ServletApiRequest.getSession(ServletApiRequest.java:530)
stirling-pdf-1  |       at jakarta.servlet.http.HttpServletRequestWrapper.getSession(HttpServletRequestWrapper.java:229)
stirling-pdf-1  |       at jakarta.servlet.http.HttpServletRequestWrapper.getSession(HttpServletRequestWrapper.java:229)
stirling-pdf-1  |       at org.springframework.security.web.csrf.HttpSessionCsrfTokenRepository.saveToken(HttpSessionCsrfTokenRepository.java:58)
stirling-pdf-1  |       at org.springframework.security.web.csrf.RepositoryDeferredCsrfToken.init(RepositoryDeferredCsrfToken.java:67)
stirling-pdf-1  |       at org.springframework.security.web.csrf.RepositoryDeferredCsrfToken.get(RepositoryDeferredCsrfToken.java:48)
stirling-pdf-1  |       at org.springframework.security.web.csrf.XorCsrfTokenRequestAttributeHandler.lambda$deferCsrfTokenUpdate$0(XorCsrfTokenRequestAttributeHandler.java:63)
stirling-pdf-1  |       at org.springframework.security.web.csrf.XorCsrfTokenRequestAttributeHandler$CachedCsrfTokenSupplier.get(XorCsrfTokenRequestAttributeHandler.java:139)
stirling-pdf-1  |       at org.springframework.security.web.csrf.XorCsrfTokenRequestAttributeHandler$CachedCsrfTokenSupplier.get(XorCsrfTokenRequestAttributeHandler.java:126)
stirling-pdf-1  |       at org.springframework.security.web.csrf.CsrfTokenRequestAttributeHandler$SupplierCsrfToken.getDelegate(CsrfTokenRequestAttributeHandler.java:89)
stirling-pdf-1  |       at org.springframework.security.web.csrf.CsrfTokenRequestAttributeHandler$SupplierCsrfToken.getParameterName(CsrfTokenRequestAttributeHandler.java:80)
stirling-pdf-1  |       at org.springframework.security.web.servlet.support.csrf.CsrfRequestDataValueProcessor.getExtraHiddenFields(CsrfRequestDataValueProcessor.java:73)
stirling-pdf-1  |       at org.thymeleaf.spring6.context.webmvc.SpringWebMvcThymeleafRequestDataValueProcessor.getExtraHiddenFields(SpringWebMvcThymeleafRequestDataValueProcessor.java:80)
stirling-pdf-1  |       at org.thymeleaf.spring6.requestdata.RequestDataValueProcessorUtils.getExtraHiddenFields(RequestDataValueProcessorUtils.java:79)
stirling-pdf-1  |       at org.thymeleaf.spring6.processor.SpringActionTagProcessor.doProcess(SpringActionTagProcessor.java:118)
stirling-pdf-1  |       at org.thymeleaf.standard.processor.AbstractStandardExpressionAttributeTagProcessor.doProcess(AbstractStandardExpressionAttributeTagProcessor.java:160)
stirling-pdf-1  |       at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74)
stirling-pdf-1  |       ... 182 common frames omitted
```


### Additional Information

_No response_

### Browsers Affected

Chrome, Microsoft Edge

### No Duplicate of the Issue

- [X] I have verified that there are no existing issues raised related to my problem.

--- a/build.gradle
+++ b/build.gradle
@@ -137,7 +137,7 @@ dependencies {
 
     if (System.getenv("DOCKER_ENABLE_SECURITY") != "false") {
         implementation "org.springframework.boot:spring-boot-starter-security:$springBootVersion"
-        runtimeOnly "org.thymeleaf.extras:thymeleaf-extras-springsecurity5:3.1.2.RELEASE"
+        implementation "org.thymeleaf.extras:thymeleaf-extras-springsecurity5:3.1.2.RELEASE"
         implementation "org.springframework.boot:spring-boot-starter-data-jpa:$springBootVersion"
         implementation "org.springframework.boot:spring-boot-starter-oauth2-client:$springBootVersion"
 

--- a/build.gradle
+++ b/build.gradle
@@ -154,6 +154,10 @@ dependencies {
         implementation "org.springframework.security:spring-security-saml2-service-provider"
 
         implementation 'com.coveo:saml-client:5.0.0'
+        // https://mvnrepository.com/artifact/org.thymeleaf.extras/thymeleaf-extras-springsecurity4
+        implementation group: 'org.thymeleaf.extras', name: 'thymeleaf-extras-springsecurity4', version: '3.0.5.RELEASE'
+
+
     }
 
     testImplementation "org.springframework.boot:spring-boot-starter-test:$springBootVersion"

--- a/build.gradle
+++ b/build.gradle
@@ -154,8 +154,6 @@ dependencies {
         implementation "org.springframework.security:spring-security-saml2-service-provider"
 
         implementation 'com.coveo:saml-client:5.0.0'
-        // https://mvnrepository.com/artifact/org.thymeleaf.extras/thymeleaf-extras-springsecurity4
-        implementation group: 'org.thymeleaf.extras', name: 'thymeleaf-extras-springsecurity4', version: '3.0.5.RELEASE'
 
 
     }

--- a/src/main/java/stirling/software/SPDF/config/security/SecurityConfiguration.java
+++ b/src/main/java/stirling/software/SPDF/config/security/SecurityConfiguration.java
@@ -34,14 +34,22 @@ import org.springframework.security.saml2.provider.service.registration.RelyingP
 import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrationRepository;
 import org.springframework.security.saml2.provider.service.web.authentication.Saml2WebSsoAuthenticationFilter;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.access.AccessDeniedHandler;
 import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 import org.springframework.security.web.authentication.rememberme.PersistentTokenRepository;
+import org.springframework.security.web.context.SecurityContextHolderFilter;
 import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+import org.springframework.security.web.csrf.CsrfFilter;
+import org.springframework.security.web.csrf.CsrfToken;
+import org.springframework.security.web.csrf.CsrfTokenRequestAttributeHandler;
+import org.springframework.security.web.csrf.InvalidCsrfTokenException;
 import org.springframework.security.web.savedrequest.NullRequestCache;
+import org.springframework.security.web.session.HttpSessionEventPublisher;
 import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.thymeleaf.extras.springsecurity5.dialect.SpringSecurityDialect;
 
+import jakarta.servlet.http.HttpServletResponse;
 import lombok.extern.slf4j.Slf4j;
-import org.thymeleaf.extras.springsecurity4.dialect.SpringSecurityDialect;
 import stirling.software.SPDF.config.security.oauth2.CustomOAuth2AuthenticationFailureHandler;
 import stirling.software.SPDF.config.security.oauth2.CustomOAuth2AuthenticationSuccessHandler;
 import stirling.software.SPDF.config.security.oauth2.CustomOAuth2UserService;
@@ -91,15 +99,23 @@ public class SecurityConfiguration {
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
 
+    	
+    	
+    	
         if (loginEnabledValue) {
             http.addFilterBefore(
                     userAuthenticationFilter, UsernamePasswordAuthenticationFilter.class);
             if (applicationProperties.getSecurity().getCsrfDisabled()) {
                 http.csrf(csrf -> csrf.disable());
             } else {
-                http.csrf(csrf -> csrf
-                        .csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
-                );
+             CookieCsrfTokenRepository cookieRepo =  CookieCsrfTokenRepository.withHttpOnlyFalse();
+           	 CsrfTokenRequestAttributeHandler requestHandler = new CsrfTokenRequestAttributeHandler();
+           	 requestHandler.setCsrfRequestAttributeName(null);
+            	http.csrf(csrf -> 
+                csrf.csrfTokenRepository(cookieRepo)
+                    .csrfTokenRequestHandler(requestHandler)
+            );
+            	
             }
             http.addFilterBefore(rateLimitingFilter(), UsernamePasswordAuthenticationFilter.class);
             http.addFilterAfter(firstLoginFilter, UsernamePasswordAuthenticationFilter.class);
@@ -111,7 +127,7 @@ public class SecurityConfiguration {
                                     .maxSessionsPreventsLogin(false)
                                     .sessionRegistry(sessionRegistry)
                                     .expiredUrl("/login?logout=true"));
-
+     
             http.authenticationProvider(daoAuthenticationProvider());
             http.requestCache(requestCache -> requestCache.requestCache(new NullRequestCache()));
             http.logout(
@@ -119,6 +135,7 @@ public class SecurityConfiguration {
                             logout.logoutRequestMatcher(new AntPathRequestMatcher("/logout"))
                                     .logoutSuccessHandler(
                                             new CustomLogoutSuccessHandler(applicationProperties))
+                                    .clearAuthentication(true)
                                     .invalidateHttpSession(true) // Invalidate session
                                     .deleteCookies("JSESSIONID", "remember-me"));
             http.rememberMe(
@@ -209,7 +226,8 @@ public class SecurityConfiguration {
             }
 
             // Handle SAML
-            if (applicationProperties.getSecurity().isSaml2Activ() && applicationProperties.getSystem().getEnableAlphaFunctionality()) {
+            if (applicationProperties.getSecurity().isSaml2Activ()
+                    && applicationProperties.getSystem().getEnableAlphaFunctionality()) {
                 http.authenticationProvider(samlAuthenticationProvider());
                 http.saml2Login(
                                 saml2 ->
@@ -228,21 +246,19 @@ public class SecurityConfiguration {
         } else {
             if (applicationProperties.getSecurity().getCsrfDisabled()) {
                 http.csrf(csrf -> csrf.disable());
+            } else {
+            	CookieCsrfTokenRepository cookieRepo =  CookieCsrfTokenRepository.withHttpOnlyFalse();
+              	 CsrfTokenRequestAttributeHandler requestHandler = new CsrfTokenRequestAttributeHandler();
+              	 requestHandler.setCsrfRequestAttributeName(null);
+               	http.csrf(csrf -> 
+                   csrf.csrfTokenRepository(cookieRepo)
+                       .csrfTokenRequestHandler(requestHandler)
+               );
             }
             http.authorizeHttpRequests(authz -> authz.anyRequest().permitAll());
         }
 
         return http.build();
-    }
-
-    @Bean
-    @ConditionalOnProperty(
-            name = "security.csrfDisabled.false",
-            havingValue = "true",
-            matchIfMissing = false)
-    //need to make conditional check for csrf disablement OR remove that option alltogether
-    public SpringSecurityDialect springSecurityDialect() {
-        return new SpringSecurityDialect();
     }
 
     @Bean

--- a/src/main/java/stirling/software/SPDF/config/security/SecurityConfiguration.java
+++ b/src/main/java/stirling/software/SPDF/config/security/SecurityConfiguration.java
@@ -34,21 +34,13 @@ import org.springframework.security.saml2.provider.service.registration.RelyingP
 import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrationRepository;
 import org.springframework.security.saml2.provider.service.web.authentication.Saml2WebSsoAuthenticationFilter;
 import org.springframework.security.web.SecurityFilterChain;
-import org.springframework.security.web.access.AccessDeniedHandler;
 import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 import org.springframework.security.web.authentication.rememberme.PersistentTokenRepository;
-import org.springframework.security.web.context.SecurityContextHolderFilter;
 import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
-import org.springframework.security.web.csrf.CsrfFilter;
-import org.springframework.security.web.csrf.CsrfToken;
 import org.springframework.security.web.csrf.CsrfTokenRequestAttributeHandler;
-import org.springframework.security.web.csrf.InvalidCsrfTokenException;
 import org.springframework.security.web.savedrequest.NullRequestCache;
-import org.springframework.security.web.session.HttpSessionEventPublisher;
 import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
-import org.thymeleaf.extras.springsecurity5.dialect.SpringSecurityDialect;
 
-import jakarta.servlet.http.HttpServletResponse;
 import lombok.extern.slf4j.Slf4j;
 import stirling.software.SPDF.config.security.oauth2.CustomOAuth2AuthenticationFailureHandler;
 import stirling.software.SPDF.config.security.oauth2.CustomOAuth2AuthenticationSuccessHandler;
@@ -99,23 +91,21 @@ public class SecurityConfiguration {
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
 
-    	
-    	
-    	
         if (loginEnabledValue) {
             http.addFilterBefore(
                     userAuthenticationFilter, UsernamePasswordAuthenticationFilter.class);
             if (applicationProperties.getSecurity().getCsrfDisabled()) {
                 http.csrf(csrf -> csrf.disable());
             } else {
-             CookieCsrfTokenRepository cookieRepo =  CookieCsrfTokenRepository.withHttpOnlyFalse();
-           	 CsrfTokenRequestAttributeHandler requestHandler = new CsrfTokenRequestAttributeHandler();
-           	 requestHandler.setCsrfRequestAttributeName(null);
-            	http.csrf(csrf -> 
-                csrf.csrfTokenRepository(cookieRepo)
-                    .csrfTokenRequestHandler(requestHandler)
-            );
-            	
+                CookieCsrfTokenRepository cookieRepo =
+                        CookieCsrfTokenRepository.withHttpOnlyFalse();
+                CsrfTokenRequestAttributeHandler requestHandler =
+                        new CsrfTokenRequestAttributeHandler();
+                requestHandler.setCsrfRequestAttributeName(null);
+                http.csrf(
+                        csrf ->
+                                csrf.csrfTokenRepository(cookieRepo)
+                                        .csrfTokenRequestHandler(requestHandler));
             }
             http.addFilterBefore(rateLimitingFilter(), UsernamePasswordAuthenticationFilter.class);
             http.addFilterAfter(firstLoginFilter, UsernamePasswordAuthenticationFilter.class);
@@ -127,7 +117,7 @@ public class SecurityConfiguration {
                                     .maxSessionsPreventsLogin(false)
                                     .sessionRegistry(sessionRegistry)
                                     .expiredUrl("/login?logout=true"));
-     
+
             http.authenticationProvider(daoAuthenticationProvider());
             http.requestCache(requestCache -> requestCache.requestCache(new NullRequestCache()));
             http.logout(
@@ -247,13 +237,15 @@ public class SecurityConfiguration {
             if (applicationProperties.getSecurity().getCsrfDisabled()) {
                 http.csrf(csrf -> csrf.disable());
             } else {
-            	CookieCsrfTokenRepository cookieRepo =  CookieCsrfTokenRepository.withHttpOnlyFalse();
-              	 CsrfTokenRequestAttributeHandler requestHandler = new CsrfTokenRequestAttributeHandler();
-              	 requestHandler.setCsrfRequestAttributeName(null);
-               	http.csrf(csrf -> 
-                   csrf.csrfTokenRepository(cookieRepo)
-                       .csrfTokenRequestHandler(requestHandler)
-               );
+                CookieCsrfTokenRepository cookieRepo =
+                        CookieCsrfTokenRepository.withHttpOnlyFalse();
+                CsrfTokenRequestAttributeHandler requestHandler =
+                        new CsrfTokenRequestAttributeHandler();
+                requestHandler.setCsrfRequestAttributeName(null);
+                http.csrf(
+                        csrf ->
+                                csrf.csrfTokenRepository(cookieRepo)
+                                        .csrfTokenRequestHandler(requestHandler));
             }
             http.authorizeHttpRequests(authz -> authz.anyRequest().permitAll());
         }

--- a/src/main/java/stirling/software/SPDF/config/security/SecurityConfiguration.java
+++ b/src/main/java/stirling/software/SPDF/config/security/SecurityConfiguration.java
@@ -36,10 +36,12 @@ import org.springframework.security.saml2.provider.service.web.authentication.Sa
 import org.springframework.security.web.SecurityFilterChain;
 import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 import org.springframework.security.web.authentication.rememberme.PersistentTokenRepository;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
 import org.springframework.security.web.savedrequest.NullRequestCache;
 import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
 
 import lombok.extern.slf4j.Slf4j;
+import org.thymeleaf.extras.springsecurity4.dialect.SpringSecurityDialect;
 import stirling.software.SPDF.config.security.oauth2.CustomOAuth2AuthenticationFailureHandler;
 import stirling.software.SPDF.config.security.oauth2.CustomOAuth2AuthenticationSuccessHandler;
 import stirling.software.SPDF.config.security.oauth2.CustomOAuth2UserService;
@@ -94,6 +96,10 @@ public class SecurityConfiguration {
                     userAuthenticationFilter, UsernamePasswordAuthenticationFilter.class);
             if (applicationProperties.getSecurity().getCsrfDisabled()) {
                 http.csrf(csrf -> csrf.disable());
+            } else {
+                http.csrf(csrf -> csrf
+                        .csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
+                );
             }
             http.addFilterBefore(rateLimitingFilter(), UsernamePasswordAuthenticationFilter.class);
             http.addFilterAfter(firstLoginFilter, UsernamePasswordAuthenticationFilter.class);
@@ -227,6 +233,16 @@ public class SecurityConfiguration {
         }
 
         return http.build();
+    }
+
+    @Bean
+    @ConditionalOnProperty(
+            name = "security.csrfDisabled.false",
+            havingValue = "true",
+            matchIfMissing = false)
+    //need to make conditional check for csrf disablement OR remove that option alltogether
+    public SpringSecurityDialect springSecurityDialect() {
+        return new SpringSecurityDialect();
     }
 
     @Bean

--- a/src/main/resources/templates/change-creds.html
+++ b/src/main/resources/templates/change-creds.html
@@ -27,7 +27,7 @@
 
               <!-- Change Username Form -->
               <h4 th:text="#{changeCreds.changePassword}">Change password</h4>
-              <form action="api/v1/user/change-password-on-login" method="post" id="formsavechangecreds">
+              <form th:action="@{'api/v1/user/change-password-on-login'}" method="post" id="formsavechangecreds">
                 <div class="mb-3">
                   <label for="currentPassword" th:text="#{changeCreds.oldPassword}">Old Password</label>
                   <input type="password" class="form-control" name="currentPassword" id="currentPassword" th:placeholder="#{changeCreds.oldPassword}">

--- a/src/main/resources/templates/fragments/navbar.html
+++ b/src/main/resources/templates/fragments/navbar.html
@@ -351,7 +351,7 @@
             </a>
             <div class="dropdown-menu dropdown-menu-tp" aria-labelledby="searchDropdown">
               <div class="dropdown-menu-wrapper px-xl-2 px-2">
-                <form class="d-flex p-2 search-form" id="searchForm">
+                <form th:action="" class="d-flex p-2 search-form" id="searchForm">
                   <input class="form-control search-input" type="search" placeholder="Search" aria-label="Search" id="navbarSearchInput">
                 </form>
                 <!-- Search Results -->

--- a/src/main/resources/templates/fragments/navbar.html
+++ b/src/main/resources/templates/fragments/navbar.html
@@ -351,7 +351,7 @@
             </a>
             <div class="dropdown-menu dropdown-menu-tp" aria-labelledby="searchDropdown">
               <div class="dropdown-menu-wrapper px-xl-2 px-2">
-                <form th:action="" class="d-flex p-2 search-form" id="searchForm">
+                <form th:action="@{''}" class="d-flex p-2 search-form" id="searchForm">
                   <input class="form-control search-input" type="search" placeholder="Search" aria-label="Search" id="navbarSearchInput">
                 </form>
                 <!-- Search Results -->

--- a/src/main/resources/templates/merge-pdfs.html
+++ b/src/main/resources/templates/merge-pdfs.html
@@ -19,7 +19,7 @@
               <span class="material-symbols-rounded tool-header-icon organize">add_to_photos</span>
               <span class="tool-header-text" th:text="#{merge.header}"></span>
             </div>
-            <form action="api/v1/general/merge-pdfs" method="post" enctype="multipart/form-data">
+            <form th:action="@{'api/v1/general/merge-pdfs'}" method="post" enctype="multipart/form-data">
               <div class="mb-3">
                 <label th:text="#{multiPdfDropPrompt}" for="fileInput-input"></label>
                 <div

--- a/src/main/resources/templates/misc/adjust-contrast.html
+++ b/src/main/resources/templates/misc/adjust-contrast.html
@@ -21,7 +21,7 @@
         <div class="container">
           <div class="row justify-content-center">
             <div class="col-md-12 bg-card">
-              <form th:action="">
+              <form th:action="@{''}">
               <div class="row justify-content-center">
 
                 <div class="col-md-3">

--- a/src/main/resources/templates/misc/adjust-contrast.html
+++ b/src/main/resources/templates/misc/adjust-contrast.html
@@ -21,7 +21,7 @@
         <div class="container">
           <div class="row justify-content-center">
             <div class="col-md-12 bg-card">
-              <form>
+              <form th:action="">
               <div class="row justify-content-center">
 
                 <div class="col-md-3">

--- a/src/main/resources/templates/misc/remove-annotations.html
+++ b/src/main/resources/templates/misc/remove-annotations.html
@@ -16,7 +16,7 @@
                 <span class="material-symbols-rounded tool-header-icon other">thread_unread</span>
                 <span class="tool-header-text" th:text="#{removeAnnotations.header}"></span>
               </div>
-              <form id="pdfForm" th:action="" class="mb-3">
+              <form id="pdfForm" th:action="@{''}" class="mb-3">
                 <div class="custom-file">
                   <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf', remoteCall='false')}"></div>
                 </div>

--- a/src/main/resources/templates/misc/remove-annotations.html
+++ b/src/main/resources/templates/misc/remove-annotations.html
@@ -16,7 +16,7 @@
                 <span class="material-symbols-rounded tool-header-icon other">thread_unread</span>
                 <span class="tool-header-text" th:text="#{removeAnnotations.header}"></span>
               </div>
-              <form id="pdfForm" class="mb-3">
+              <form id="pdfForm" th:action="" class="mb-3">
                 <div class="custom-file">
                   <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf', remoteCall='false')}"></div>
                 </div>

--- a/src/main/resources/templates/remove-image-pdf.html
+++ b/src/main/resources/templates/remove-image-pdf.html
@@ -20,7 +20,7 @@
                         <span class="material-symbols-rounded tool-header-icon word">remove_selection</span>
                         <span class="tool-header-text" th:text="#{removeImage.header}"></span>
                     </div>
-                        <form action="api/v1/general/remove-image-pdf" method="post" enctype="multipart/form-data">
+                        <form th:action="@{'api/v1/general/remove-image-pdf'}" method="post" enctype="multipart/form-data">
                         <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf')}"></div>
 
                         <br>

--- a/src/main/resources/templates/security/add-password.html
+++ b/src/main/resources/templates/security/add-password.html
@@ -16,7 +16,7 @@
                 <span class="material-symbols-rounded tool-header-icon security">lock</span>
                 <span class="tool-header-text" th:text="#{addPassword.header}"></span>
               </div>
-              <form action="api/v1/security/add-password" method="post" enctype="multipart/form-data">
+              <form th:action="@{'api/v1/security/add-password'}" method="post" enctype="multipart/form-data">
                 <div class="mb-3">
                   <label th:text="#{addPassword.selectText.1}"></label>
                   <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf')}"></div>

--- a/src/main/resources/templates/security/add-watermark.html
+++ b/src/main/resources/templates/security/add-watermark.html
@@ -17,7 +17,7 @@
                 <span class="tool-header-text" th:text="#{watermark.header}"></span>
               </div>
 
-              <form method="post" enctype="multipart/form-data" action="api/v1/security/add-watermark">
+              <form method="post" enctype="multipart/form-data" th:action="@{'api/v1/security/add-watermark'}">
                 <div class="mb-3">
                   <label th:text="#{watermark.selectText.1}"></label>
                   <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf')}"></div>

--- a/src/main/resources/templates/security/auto-redact.html
+++ b/src/main/resources/templates/security/auto-redact.html
@@ -16,7 +16,7 @@
                 <span class="material-symbols-rounded tool-header-icon security">ink_eraser</span>
                 <span class="tool-header-text" th:text="#{autoRedact.header}"></span>
               </div>
-              <form action="api/v1/security/auto-redact" method="post" enctype="multipart/form-data">
+              <form th:action="@{'api/v1/security/auto-redact'}" method="post" enctype="multipart/form-data">
                 <div class="mb-3">
                   <input type="file" class="form-control" id="fileInput" name="fileInput" required accept="application/pdf">
                 </div>

--- a/src/main/resources/templates/security/cert-sign.html
+++ b/src/main/resources/templates/security/cert-sign.html
@@ -16,7 +16,7 @@
                 <span class="material-symbols-rounded tool-header-icon security">workspace_premium</span>
                 <span class="tool-header-text" th:text="#{certSign.header}"></span>
               </div>
-              <form action="api/v1/security/cert-sign" method="post" enctype="multipart/form-data">
+              <form th:action="@{'api/v1/security/cert-sign'}" method="post" enctype="multipart/form-data">
                 <div class="mb-3">
                   <label th:text="#{certSign.selectPDF}"></label>
                   <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf')}"></div>

--- a/src/main/resources/templates/security/change-permissions.html
+++ b/src/main/resources/templates/security/change-permissions.html
@@ -17,7 +17,7 @@
                 <span class="tool-header-text" th:text="#{permissions.header}"></span>
               </div>
               <p th:text="#{permissions.warning}"></p>
-              <form action="api/v1/security/add-password" method="post" enctype="multipart/form-data">
+              <form th:action="@{'api/v1/security/add-password'}" method="post" enctype="multipart/form-data">
                 <div class="mb-3">
                   <label th:text="#{permissions.selectText.1}"></label>
                   <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf')}"></div>

--- a/src/main/resources/templates/security/remove-cert-sign.html
+++ b/src/main/resources/templates/security/remove-cert-sign.html
@@ -16,7 +16,7 @@
                 <span class="material-symbols-rounded tool-header-icon security">remove_moderator</span>
                 <span class="tool-header-text" th:text="#{removeCertSign.header}"></span>
               </div>
-              <form action="api/v1/security/remove-cert-sign" method="post" enctype="multipart/form-data">
+              <form th:action="@{'api/v1/security/remove-cert-sign'}" method="post" enctype="multipart/form-data">
                 <div class="mb-3">
                   <label th:text="#{removeCertSign.selectPDF}"></label>
                   <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf')}"></div>

--- a/src/main/resources/templates/security/remove-password.html
+++ b/src/main/resources/templates/security/remove-password.html
@@ -16,7 +16,7 @@
                 <span class="material-symbols-rounded tool-header-icon security">lock_open_right</span>
                 <span class="tool-header-text" th:text="#{removePassword.header}"></span>
               </div>
-              <form action="api/v1/security/remove-password" method="post" enctype="multipart/form-data">
+              <form th:action="@{'api/v1/security/remove-password'}" method="post" enctype="multipart/form-data">
                 <div class="mb-3">
                   <label th:text="#{removePassword.selectText.1}"></label>
                   <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf')}"></div>

--- a/src/main/resources/templates/security/remove-watermark.html
+++ b/src/main/resources/templates/security/remove-watermark.html
@@ -16,7 +16,7 @@
                 <span class="material-symbols-rounded tool-header-icon security">water_drop</span>
                 <span class="tool-header-text" th:text="#{remove-watermark.header}"></span>
               </div>
-              <form method="post" enctype="multipart/form-data" action="api/v1/security/remove-watermark">
+              <form method="post" enctype="multipart/form-data" th:action="@{'api/v1/security/remove-watermark'}">
                 <div class="mb-3">
                   <label th:text="#{remove-watermark.selectText.1}"></label>
                   <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf')}"></div>

--- a/src/main/resources/templates/security/sanitize-pdf.html
+++ b/src/main/resources/templates/security/sanitize-pdf.html
@@ -16,7 +16,7 @@
                 <span class="material-symbols-rounded tool-header-icon security">sanitizer</span>
                 <span class="tool-header-text" th:text="#{sanitizePDF.header}"></span>
               </div>
-              <form action="api/v1/security/sanitize-pdf" method="post" enctype="multipart/form-data">
+              <form th:action="@{'api/v1/security/sanitize-pdf'}" method="post" enctype="multipart/form-data">
                 <div class="mb-3">
                   <div th:replace="~{fragments/common :: fileSelector(name='fileInput', multipleInputsForSingleRequest=false, accept='application/pdf')}"></div>
                 </div>

